sktree/ensemble/_honest_forest.py

```diff
@@ -0,0 +1,497 @@
+    oob_score_ : float
+        Score of the training dataset obtained using an out-of-bag estimate.
+        This attribute exists only when ``oob_score`` is True.
+
+    oob_decision_function_ : ndarray of shape (n_samples, n_classes) or \
+            (n_samples, n_classes, n_outputs)
+        Decision function computed with out-of-bag estimate on the training
+        set. If n_estimators is small it might be possible that a data point
+        was never left out during the bootstrap. In this case,
+        `oob_decision_function_` might contain NaN. This attribute exists
+        only when ``oob_score`` is True.
+
+    honest_decision_function_ : ndarray of shape (n_samples, n_classes) or \
+            (n_samples, n_classes, n_outputs)
+        Decision function computed on each sample, including only the trees
+        for which it was in the honest subsample. It is possible that a sample
+        is never in the honest subset in which case `honest_decision_function_`
+        might contain NaN. This attribute exists when `honest_score_` is True.
+
+    structure_indices_ : list of lists, shape=(n_estimators, n_structure)
+        Indices of training samples used to learn the structure
+
+    honest_indices_ : list of lists, shape=(n_estimators, n_honest)
+        Indices of training samples used to learn leaf estimates
+
+    Notes
+    -----
+    The default values for the parameters controlling the size of the trees
+    (e.g. ``max_depth``, ``min_samples_leaf``, etc.) lead to fully grown and
+    unpruned trees which can potentially be very large on some data sets. To
+    reduce memory consumption, the complexity and size of the trees should be
+    controlled by setting those parameter values.
+
+    The features are always randomly permuted at each split. Therefore,
+    the best found split may vary, even with the same training data,
+    ``max_features=n_features`` and ``bootstrap=False``, if the improvement
+    of the criterion is identical for several splits enumerated during the
+    search of the best split. To obtain a deterministic behaviour during
+    fitting, ``random_state`` has to be fixed.
+
+    References
+    ----------
+    .. [1] L. Breiman, "Random Forests", Machine Learning, 45(1), 5-32, 2001.
+
+    .. [2] S. Athey, J. Tibshirani, and S. Wager. "Generalized
+            Random Forests", Annals of Statistics, 2019.
+
+    Examples
+    --------
+    >>> from honest_forests.estimators import HonestForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> X, y = make_classification(n_samples=1000, n_features=4,
+    ...                            n_informative=2, n_redundant=0,
+    ...                            random_state=0, shuffle=False)
+    >>> clf = HonestForestClassifier(max_depth=2, random_state=0)
+    >>> clf.fit(X, y)
+    HonestForestClassifier(...)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
+    [1]
+    """
+
+    def __init__(
+        self,
+        n_estimators=100,
```

Comment:
We may want to allow the passing in of a tree estimator with the default being decisiontreeclassifier?